src/io/read_tree_lhalo_hdf5.c

```diff
@@ -330,19 +343,31 @@ int64_t load_forest_hdf5(const int64_t forestnr, struct halo_data **halos, struc
     H5Dclose(dataset);
     const int64_t nhalos = dims[0];
 
+    status = read_dataset_shape(fd, dataset_name, &ndims, &dims);
+    if(status != EXIT_SUCCESS) {
+        const int neg_status = status < 0 ? status:-status;
+        return neg_status;
+    }
+
+    XRETURN(ndims == 0, -1, "Error: Expected field = '%s' to be 1-D array with ndims == 0. Instead found ndims = %d\n",
+            field_name, ndims);
+    const int64_t nhalos = dims[0];
+    free(dims);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Deallocation of an auto-variable results in undefined behaviour.](https://app.codacy.com/manual/jacobseiler/sage-model/pullRequest?prid=4620074)